internal/uvm/types.go

```diff
@@ -131,3 +142,35 @@ type UtilityVM struct {
 	// uvm.OptionsLCOW or uvm.OptionsWCOW
 	createOpts interface{}
 }
+
+// refCounter keeps track of object's reference counts
+type refCounter interface {
+	RefCount() uint32
+	AddRef() uint32
+	RemoveRef() uint32
+}
+
+// genericDevice represents a device that can be mounted onto a guest
+type genericDevice interface {
+	MountPath() string
+	HostPath() string
+	ResourcePath() string
+	DeviceNumber() uint32
+	DeviceSize() uint64
+	// device settings as seen by the host
+	HostSettings() interface{}
+	// device settings as seen by the guest
+	GuestSettings() interface{}
+}
```

Comment:
I don't really feel like these interfaces are really needed. Was there a specific reason for them?

Reply:
Oh I see now (immediately after I commented this :D) I understand, but I'm also not sure I like the fact that have have basically a four-times chained interface for the mappedDevice. I think it would be fine to have just mappedDevice and generic device and move around some of the other calls if you'd like. What do you think?

Reply:
Was the plan to make the other refcount based resources (scsi, vpci, vsmb etc.) implement refCounter?

Reply:
@katiewasnothere , for this PR we didn't really even need these interfaces. this is something I did to give an idea of what I was talking about with regards to potential rework/refactor. as @dcantah pointed out the idea indeed was that we have some sort of generic device interface (the one in this PR probably still not generic enough) and have other devices implement those (e.g. `refCounter`). since you both are definitely more familiar with the code base, I'd like to hear your inputs on this

Reply:
What devices did you have in mind for implementing only `refCounter`?

Reply:
But in that case, I think we should just move the refcounting funcs into the `genericDevice` interface. I don't see a case for anything to implement only `refCounter` and be worth it.